### PR TITLE
Enterprise switcher and navigation check

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Created a dropdown in Enterprise management page for switching between enterprises for #421. Displays dropdown list of enterprises the current user can edit, takes the user to the selected Edit Enterprise page when selected.

Also addressed issue #420, Edit Enterprise pages now display a confirmation popup if the user tries to navigate away from the page whilst there are unsaved changes on the form.

### CODE REVIEW

File: app/assets/javascripts/admin/utils/services/navigation_check.js.coffee [hunk 1]
```diff
@@ -1,5 +1,5 @@
 angular.module("admin.utils")
-  .factory "NavigationCheck", ($window, $rootScope) ->
+  .factory "NavigationCheck", ($window, $rootScope, FormState) ->
```
Comment (mkllnk) [inline app/assets/javascripts/admin/utils/services/navigation_check.js.coffee]: FormState is actually not used here.

**Comment by mkllnk** [comment]
@Matt-Yorkley I did some improvements in #1369. What do you think? And Sally found a little issue in https://github.com/openfoodfoundation/openfoodnetwork/issues/421#issuecomment-271772095. Once that's fixed, we can merge it, I think.

**Comment by mkllnk** [comment]
@Matt-Yorkley Nice improvements. I actually just realised that I didn't upload my re-factoring before I opened my pull request. What do you think about https://github.com/openfoodfoundation/openfoodnetwork/pull/1369/commits/91996b122f2167fb4ac9fc77904e70a355f2dacb?

**Comment by Matt-Yorkley** [comment]
@mkllnk Looks nice and tidy :) Are there any other files that reference that FormState file?

**Comment by mkllnk** [comment]
@Matt-Yorkley 

> Are there any other files that reference that FormState file?

No, I can't find any other references. Are you happy to merge it into your branch?

**Comment by Matt-Yorkley** [comment]
@mkllnk Sure. What's the best way to do that? You want to submit a PR to my fork?

**Comment by Matt-Yorkley** [comment]
@mkllnk I've merged your changes. Does it need a final pass from testers or is it good to go?

**Comment by Matt-Yorkley** [comment]
@mklink @oeoeaio can we merge this now? I think it's all done with testing and review. I'm working on the product import and the E2E permissions scope I added in this PR would be really useful...

**Comment by sstead** [comment]
420 and 421 working well.
421 - Works in SuperAdmin and normal users. Updates according to E2Es. Can type to search or scroll through list. Updates if enterprise is added that's owned/managed by user.
Small nice to have improvement - if user has only 1 enterprise don't show the search/scroll option in 421.
Ready to push @mkllnk

**Pull request merged**